### PR TITLE
feat: enable semantic release for docker builds

### PR DESCRIPTION
**Pull request by garethjevans**

### CODE REVIEW

**Comment by garethjevans** [comment]
Depends on https://github.com/jenkinsci/JenkinsPipelineUnit/pull/343

File: vars/buildDockerAndPublishImage.groovy [hunk 1]
```diff
@@ -73,11 +91,35 @@ def call(String imageName, Map config=[:]) {
...
+          allOf {
+            expression { env.SEMANTIC_RELEASE == 'true' }
+            branch dockerConfig.mainBranch  
+          }
+        }
+        steps {
+          echo "Configuring credential.helper"
+          sh 'git config --local credential.helper "!f() { echo username=\\$GIT_USERNAME; echo password=\\$GIT_PASSWORD; }; f"'
+
+          withCredentials([usernamePassword(credentialsId: "${dockerConfig.gitCredentials}", passwordVariable: 'GIT_PASSWORD', usernameVariable: 'GIT_USERNAME')]) {
```
Comment (dduportal) [inline vars/buildDockerAndPublishImage.groovy:105]: Is there a reason to not use the environment form?

e.g.

```
stage("Semantic Release") {
  when {
    allOf {
      expression { env.SEMANTIC_RELEASE == 'true' }
      branch dockerConfig.mainBranch  
    }
  }
  environment {
    GITHUB_CREDENTIALS = credential("${dockerConfig.gitCredentials}")
  }
  steps {
    echo "Configuring credential.helper"
    sh 'git config --local credential.helper "!f() { echo username=\\$GITHUB_CREDENTIALS_USR; echo password=\\$GITHUB_CREDENTIALS_PSW; }; f"'
    echo "Tagging New Version: $nextVersion"
    sh "git tag $nextVersion"

    echo "Pushing Tag"
    sh "git push origin --tags"
  }
}
```

Comment (garethjevans) [reply vars/buildDockerAndPublishImage.groovy:105]: Yeah, only cause AFAIK the environment form is not implemented in JenkinsPipelineUnit.

**Review by dduportal — APPROVED** [review]
LGTM 👍

**Pull request merged**